### DISCUSSION TITLE
Waiting for an action server to become available...

### DISCUSSION THREAD

**Discussion by Illyvia** [iRobotEducation/create3_docs#308] [Questions] [answered]
Hello, I'm having trouble sending actions to the create3. I'm using a raspberry pi 400, running Ubuntu 22.04
```
pi@ubuntu:~$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
pi@ubuntu:~$ ros2 topic list
/create/battery_state
/create/cmd_audio
/create/cmd_lightring
/create/cmd_vel
/create/dock
/create/hazard_detection
/create/imu
/create/interface_buttons
/create/ir_intensity
/create/ir_opcode
/create/kidnap_status
/create/mobility_monitor/transition_event
/create/mouse
/create/odom
/create/robot_state/transition_event
/create/slip_status
/create/static_transform/transition_event
/create/stop_status
/create/tf
/create/tf_static
/create/wheel_status
/create/wheel_ticks
/create/wheel_vels
/parameter_events
/rosout
pi@ubuntu:~$ ros2 action send_goal /dock irobot_create_msgs/action/Dock "{}"
Waiting for an action server to become available...
```
I've also tried this on a virtual machine running Ubuntu 20.04.5 LTS

```
ros@ROS2-VirtualBox:~$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.5 LTS"
ros@ROS2-VirtualBox:~$ ros2 topic list
/create/battery_state
/create/cmd_audio
/create/cmd_lightring
/create/cmd_vel
/create/dock
/create/hazard_detection
/create/imu
/create/interface_buttons
/create/ir_intensity
/create/ir_opcode
/create/kidnap_status
/create/mobility_monitor/transition_event
/create/mouse
/create/odom
/create/robot_state/transition_event
/create/slip_status
/create/static_transform/transition_event
/create/stop_status
/create/tf
/create/tf_static
/create/wheel_status
/create/wheel_ticks
/create/wheel_vels
/parameter_events
/rosout
ros@ROS2-VirtualBox:~$ ros2 action send_goal /dock irobot_create_msgs/action/Dock "{}"
Waiting for an action server to become available...
```
I've also tried this with a lot of other actions.

**Comment by alsora** [comment, marked as answer]
What does `ros2 action list` returns?
Which version of ROS 2 are you using on the create 3 and on the other machines?

**Reply by Illyvia** [reply]
ros2 action list returns:
 ```
/create/audio_note_sequence
/create/dock
/create/drive_arc
/create/drive_distance
/create/led_animation
/create/navigate_to_position
/create/rotate_angle
/create/undock
/create/wall_follow
/dock
```
And how do I know what kind of ros2 is running?
The create 3 is running firmware version G.3.1

**Reply by alsora** [reply]
Your create 3 robot is using ROS 2 G (Galactic).
It's likely that your 22.04 RPi is using Humble and your 20.04  virtual box is using Galactic.

Moreover, I see that your Create 3 is using a namespace (`create`).
This means that your action goal command is wrong: it should be:

```
ros2 action send_goal /create/dock irobot_create_msgs/action/Dock "{}"
```

**Reply by Illyvia** [reply]
No, sorry. That still didn't work.

**Reply by alsora** [reply]
Can you try a different action? For example 

```
ros2 action send_goal /create/drive_distance irobot_create_msgs/action/DriveDistance "{distance: 0.5,max_translation_speed: 0.15}"
```

I would also be interested in seeing the full output of
```
ros2 action list -v
```
and
```
ros2 action info /create/dock -v
```

**Reply by Illyvia** [reply]
`ros2 action send_goal /create/drive_distance irobot_create_msgs/action/DriveDistance "{distance: 0.5,max_translation_speed: 0.15}"` Worked!!!!! `ros2 action list -v` returns:
```
 usage: ros2 [-h] [--use-python-default-buffering] Call `ros2 <command> -h` for more detailed usage. ...
ros2: error: unrecognized arguments: -v
```
and `ros2 action info /create/dock -v` returns: 
```
usage: ros2 [-h] [--use-python-default-buffering] Call `ros2 <command> -h` for more detailed usage. ...
ros2: error: unrecognized arguments: -v
```

**Reply by alsora** [reply]
Great!
Sorry, I made a mistake in the other commands, just try
```
ros2 action list --count --show-types
```

**Reply by Illyvia** [reply]
`ros2 action list --count --show-types` just returns `9`

**Reply by alsora** [reply]
And this ? 
``` 
ros2 action list --show-types
```

**Reply by Illyvia** [reply]
That returns: 
```
/create/audio_note_sequence [irobot_create_msgs/action/AudioNoteSequence]
/create/dock [irobot_create_msgs/action/DockServo]
/create/drive_arc [irobot_create_msgs/action/DriveArc]
/create/drive_distance [irobot_create_msgs/action/DriveDistance]
/create/led_animation [irobot_create_msgs/action/LedAnimation]
/create/navigate_to_position [irobot_create_msgs/action/NavigateToPosition]
/create/rotate_angle [irobot_create_msgs/action/RotateAngle]
/create/undock [irobot_create_msgs/action/Undock]
/create/wall_follow [irobot_create_msgs/action/WallFollow]
```

**Reply by alsora** [reply]
The command to invoke the dock action should then be
```
ros2 action send_goal /create/dock irobot_create_msgs/action/DockServo "{}"
```

Note how the action name is `/create/dock` and the action type is `DockServo`

**Reply by Illyvia** [reply]
That returns: `The passed action type is invalid`

**Reply by shamlian** [reply]
Not to butt in here, but I think this most recent problem with an invalid action type is because you have Humble messages installed on the machine upon which you're typing, while your robot is running Galactic.

**Reply by alsora** [reply]
Yes, as @shamlian said.
Make sure that you are interacting with the robot from a workstation with ROS 2 Galactic installed (I assume the 20.04 you mentioned above).

Communication between different ROS 2 distributions is not guaranteed to work.

**Reply by Illyvia** [reply]
Thanks so much!!!! I tried again on the virtual machine and it worked!!!

**Comment by iansexton1** [comment]
I’m having a similar issue after installing the Ubuntu 20.04 OS onto my Machine and going through all the setup steps. I confirmed communication to the robot by running the “ros 2 topic list” and “ros 2 action list” commands and got the below results. I then entered the undocking command, **ros2 action send_goal /undock irobot_create_msgs/action/Undock "{}** but the robot did not respond and got stuck “Waiting for an action server…”. I’ve already confirmed that I’m using the ‘Cyclone DDS” RMW.
![Screenshot from 2023-01-29 01-05-51](https://user-images.githubusercontent.com/109889851/215313690-f95f307b-debd-4e74-98be-d9146c63cb55.png)
![Screenshot from 2023-01-29 01-20-23](https://user-images.githubusercontent.com/109889851/215313691-74cb4f37-d946-4d55-b58f-f135c51a5c63.png)
![Screenshot from 2023-01-29 01-53-13](https://user-images.githubusercontent.com/109889851/215313692-f645521d-0169-4b52-bf7a-a2bb75164e38.png)
![Screenshot from 2023-01-29 02-59-47](https://user-images.githubusercontent.com/109889851/215313693-eff13a89-cd84-4a52-932f-9a5d6745ed89.png)

**Reply by alsora** [reply]
Hi @iansexton1, as the output of your `ros2 action list` command shows, there isn't an action named `/undock` on your robot, but rather it's named `/Nexton/undock`.

Whenever you add a ROS namespace, then you need to remember to include it when sending commands or listening to topics.

The correct way to invoke the action is as follows:

```
ros2 action send_goal /Nexton/undock irobot_create_msgs/action/Undock "{}" 
```
